### PR TITLE
Update to new macro url syntax

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @steveklabnik

### CODE REVIEW

File: src/libcore/fmt/mod.rs [hunk 1]
```diff
@@ -803,7 +803,7 @@ pub trait UpperExp {
 /// assert_eq!(output, "Hello world!");
 /// ```
 ///
-/// [write_macro]: ../../std/macro.write!.html
+/// [write_macro]: ../../std/macro.write.html
```
Comment (Mark-Simulacrum) [inline src/libcore/fmt/mod.rs]: Should this be changed to "[`write!`]" instead of [write_macro] while we're here?

**Review by Mark-Simulacrum — APPROVED** [review]

File: src/libcore/fmt/mod.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/libcore/fmt/mod.rs]: Why not!

**Comment by GuillaumeGomez** [comment]
Next, I think about removing the actual compatibility with the old url macro syntax. If people have opinion?

**Comment by steveklabnik** [comment]
> Next, I think about removing the actual compatibility with the old url macro syntax. If people have opinion?


we should leave it in for a little while

**Comment by GuillaumeGomez** [comment]
Ok. So I guess this is ready to get merged?

**Comment by GuillaumeGomez** [comment]
@steveklabnik: ping

**Pull request merged**